README.md

```diff
@@ -125,8 +134,22 @@ Current state | Next word | Next state
 
 ## A2.4
 
-«replace this with your answer»
 
+Current state | Next word | Next state
+--------------|-----------|-----------
+    S0        |    the    |     S1
+    S1        |    lazy   |     S1
+    S1        |    smelly |     S1
+    S1        |    ""     |     S2
+    S1        |    lazy   |     S2
+    S1        |    smelly |     S2
+    S2        |    cat    |     S3
+    S2        |    dog    |     S3
+    S3        |    ran    |     S4
+    S3        |    ate    |     S4
+    S3        |    ""     |     END
+    S4        |    slowly |     END
+    S4        |    noisily|     END
```

Comment:
Look at your diagram (which is perfect).

In state S1, seeing "smelly" or "lazy" takes you back to S1;  "cat" or "dog" stakes you to S2. The table for this is:

Current State | Next Word | Next State
------------- | --------- | ----------
S0            | "the"     | S1
S1            | "lazy"    | S1
S1            | "smelly"  | S1
S1            | "dog"     | S2
S1            | "cat"     | S2

Your table has an extra state S2, and your S3 is what your diagram calls S2. The two don't seem to match at all.

# -4